### PR TITLE
dailywork-ftschindler

### PR DESCRIPTION
**Pull request by ftalbrecht**

### CODE REVIEW

File: dune/xt/la/container/vector-interface.hh [hunk 1]
```diff
@@ -156,12 +156,7 @@ public:
     return this->as_imp()[ii];
   }
 
-  /**
-   * \brief   The dimension of the vector.
-   * \return  The dimension of the vector.
-   * \see     size()
-   */
-  inline size_t dim() const
+  inline size_t DUNE_DEPRECATED_MSG("Use size() instead (09.04.2018)!") dim() const
```
Comment (renemilk) [inline dune/xt/la/container/vector-interface.hh]: Do you mind using the new macro for deprecations? https://github.com/dune-community/dune-xt-common/issues/13

**Comment by ftalbrecht** [comment]
This is blocked by #41

**Comment by ftalbrecht** [comment]
There are some errors regarding our new deprecation macro, but only with clang, e.g. https://travis-ci.org/dune-community/dune-xt-la/jobs/397094406#L1215

Any thoughts, @renemilk? Am I using this wrong or is clang wrong?

**Comment by renemilk** [comment]
See https://github.com/dune-community/dune-xt-common/commit/593c8410cc651ce3c0a1476d3fa03c6c29f7e70b#diff-6ec781bfbc854755e0e4ddb31cceef7bR17

**Comment by ftalbrecht** [comment]
So I am using this wrong? :) I have a member function
```c++
inline size_t DXT_DEPRECATED_MSG("Use size() instead (09.04.2018)!") dim() const
```
and it should be
```c++
DXT_DEPRECATED_MSG("Use size() instead (09.04.2018)!") inline size_t dim() const
```
right? The latter is much more readable, btw.

**Comment by renemilk** [comment]
Yeah, that should work, at least according to what the docs said from which I made the examples.

**Pull request merged**